src/Microsoft.ML.Featurizers/RollingWindowFeaturizer.cs

```diff
@@ -0,0 +1,1028 @@
+                MaxWindowSize = maxWindowSize,
+                MinWindowSize = minWindowSize,
+                WindowCalculation = windowCalculation
+            };
+
+            return new RollingWindowEstimator(CatalogUtils.GetEnvironment(catalog), options);
+        }
+    }
+
+    /// <summary>
+    /// RollingWindow featurizer performs a rolling calculation over a window of data per grain. The currently supported window calculations are
+    /// mean, min, and max. This also adds annotations to the output column to track the min/max window sizes, as well as what calculation was performed. The horizon
+    /// is an initial offset, and the window calculation is performed starting at that initial offset, and then looping -1 until that offset is equal to 1.
+    /// </summary>
+    /// <remarks>
+    /// <format type="text/markdown"><![CDATA[
+    ///
+    /// ###  Estimator Characteristics
+    /// |  |  |
+    /// | -- | -- |
+    /// | Does this estimator need to look at the data to train its parameters? | No |
+    /// | Input column data type | double |
+    /// | Output column data type | vector of double. Size of vector is equal to the horizon |
+    ///
+    /// The <xref:Microsoft.ML.Transforms.RollingWindowEstimator> is a trivial estimator and doesn't need training.
+    ///
+    ///
+    /// ]]>
+    /// </format>
+    /// </remarks>
+    /// <seealso cref="RollingWindowExtensionClass.RollingWindow(TransformsCatalog, string[], string, RollingWindowEstimator.RollingWindowCalculation, UInt32, UInt32, UInt32, string)"/>
+    /// <seealso cref="RollingWindowExtensionClass.RollingWindow(TransformsCatalog, string[], InputOutputColumnPair[], RollingWindowEstimator.RollingWindowCalculation, UInt32, UInt32, UInt32)"/>
+    public class RollingWindowEstimator : IEstimator<RollingWindowTransformer>
+    {
+        private Options _options;
+        private readonly IHost _host;
+
+        #region Options
+
+        internal sealed class Column : OneToOneColumn
+        {
+            internal static Column Parse(string str)
+            {
+                Contracts.AssertNonEmpty(str);
+
+                var res = new Column();
+                if (res.TryParse(str))
+                    return res;
+                return null;
+            }
+
+            internal bool TryUnparse(StringBuilder sb)
+            {
+                Contracts.AssertValue(sb);
+                return TryUnparseCore(sb);
+            }
+        }
+
+        internal sealed class Options: TransformInputBase
+        {
+            [Argument((ArgumentType.MultipleUnique | ArgumentType.Required), HelpText = "List of grain columns",
+                Name = "GrainColumns", ShortName = "grains", SortOrder = 0)]
+            public string[] GrainColumns;
+
+            [Argument(ArgumentType.Multiple | ArgumentType.Required, HelpText = "New column definition (optional form: name:src)",
+                Name = "Columns", ShortName = "col", SortOrder = 1)]
```

Comment:
>Columns [](start = 24, length = 7)

Make sure its a "Column", singular. This is a standard, see other ones like ToString or RobustScaler. Unfortunately putting plural is not handled down the line in code generation in NimbusML